### PR TITLE
doc: Add project CONTRIBUTING repo page

### PR DESCRIPTION
**Pull request by dbkinder**
Summarize the ACRN contributing guidelines (fully documented in a
separate document) and add to the top level of our project repo (as
expected by GitHub and our Technical Steering Committee)

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
+Contribution Guidelines
+#######################
+
+As an open-source project, we welcome and encourage the community to submit
+patches directly to the project.  In our collaborative open source environment,
```
Comment (fitchbe) [inline CONTRIBUTING.rst:5]: "open-source" (with a hyphen) when it's a modifier (see the first sentence).

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+* ACRN development workflow is supported on Linux.
+
+* Source code for the project is maintained in the GitHub repo:
+  https://github.com/projectacrn/acrn-hypervisor
+
+* Issue and feature tracking is done using GitHub issues in this repo.
+
+* A Continuous Integration (CI) system runs on every Pull Request (PR)
+  to verify several aspects of the PR including Git commit formatting,
+  Coding Style, sanity checks builds, and documentation builds.
```
Comment (fitchbe) [inline CONTRIBUTING.rst:33]: No reason to capitalize "coding style". Also: "sanity checks builds" → "sanity-check builds".

**Review by fitchbe — APPROVED** [review]

**Comment by dbkinder** [comment]
@fitchbe thanks for the review.

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+* Source code for the project is maintained in the GitHub repo:
+  https://github.com/projectacrn/acrn-hypervisor
+
+* Issue and feature tracking is done using GitHub issues in this repo.
+
+* A Continuous Integration (CI) system runs on every Pull Request (PR)
+  to verify several aspects of the PR including Git commit formatting,
+  coding style, sanity-check builds, and documentation builds.
+
+* The `ACRN dev mailing list`_ is a great place to engage with the
```
Comment (gvancuts) [inline CONTRIBUTING.rst:35]: I would recommend we point people to the `acrn-users` mailing list instead here. It is more appropriate for a first contact and initial debugging of user issues. `acrn-dev` is a high-volume mailing list and is primarily used to submit and review patches, that can be a bit overwhelming to newcomers ;-)

We should also, at the same time, make a similar change in the contribution guidelines doc - change this sentence:
```
You should also read through discussions in the ACRN-dev mailing list to see what’s been reported on or discussed. 
```
to
```
You should also read through discussions in the ACRN-users mailing list to see what’s been reported on or discussed. 
```

**Review by gvancuts — CHANGES_REQUESTED** [review]

**Comment by dbkinder** [comment]
@wenlingz @NanlinXie I suspect I can ignore the YOCTO build failure, but what's the LDRA_CHECK failure?

**Comment by dbkinder** [comment]
OK to verify doc

**Comment by dbkinder** [comment]
OK to verify

**Pull request merged**